osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -18,14 +18,17 @@ public abstract class WorkingBeatmap : IDisposable
 
         public readonly BeatmapSetInfo BeatmapSetInfo;
 
+        public readonly BeatmapMetadata Metadata;
+
         public readonly Bindable<IEnumerable<Mod>> Mods = new Bindable<IEnumerable<Mod>>(new Mod[] { });
 
         public readonly bool WithStoryboard;
 
-        protected WorkingBeatmap(BeatmapInfo beatmapInfo, BeatmapSetInfo beatmapSetInfo, bool withStoryboard = false)
+        protected WorkingBeatmap(BeatmapInfo beatmapInfo, BeatmapSetInfo beatmapSetInfo = null, BeatmapMetadata metadata = null, bool withStoryboard = false)

```

Comment:
I don't get it. What's the point of these ctor arguments? I feel we should be going the other way and removing the setinfo argument, not adding more on top of it?